src/Task/Base/Exec.php

```diff
@@ -98,8 +99,9 @@ public function background($arg = true)
      */
     protected function getCommandDescription()
     {
-        return $this->getCommand();
```

Comment:
Now I am wondering if this PR is in fact breaking, since you have changed the contract on `getCommand()`. This isn't just relevant for unit tests; Robo uses `getCommand()` for a number of things, and if this method starts returning a string that contains placeholder instead of a string with escaped arguments, then code that called getCommand and passed it to some process execution function would fail.

Would it work out to make `getCommandDescription()` continue to call / return `getCommand()`, have `getCommand()` call `replacePlaceholders`, and add a new method `getCommandTemplate()` that returns the command string with placeholders, and use that in `Task\Exec::run()`?

Reply:
So I've just tested outside of taskExec, and it does break on other commands : most command include a custom `run()` function, so placeholders don't get replaced.
Something I've noticed on other commands is that `getCommand()` is often used to display the command to be executed, when `getCommandDescription()` should be used instead.

In any case, there's more than `Exec::run()` that needs replacing, but every run() method that uses the `ExecOnCommand` trait, since they all get their arguments from `CommandArguments` and need to have their placeholders replaced.
As far as I know, only `Exec` has a custom process creation, every other calling the `ExecCommand::executeCommand()` method. It would simply be a matter of overwriting it in an `ExecOneCommand::executeCommand()` method that would handle placeholders.

So to sum things up :
- Add a `getCommandTemplate()` method ;
- Change `ExecOneCommand::executeOneCommand()` to replace placeholders.

Sounds good?

Reply:
Yes, that sounds good.